test/test_enforced_encryption.cpp

```diff
@@ -382,19 +432,15 @@ class TestEnforcedEncryption
                     }
                 }
             }
-        });
 
-        const int connect_ret = srt_connect(m_caller_socket, psa, sizeof sa);
-        EXPECT_EQ(connect_ret, expect.connect_ret);
```

Comment:
Moving this part up looks like reviving the old version of the test with higher likelihood of failing due to missing socket closure.

Reply:
I think this whole change for this file deserves to get into a different PR. Even though this test was being revised during the works on this.